src/checks/checker.cc

```diff
@@ -367,6 +367,7 @@ void checker::run(
 
   // Get current host macros.
   nagios_macros macros;
+  memset(&macros, 0, sizeof(macros));
```

Comment:
this is a bug. these memset are dangerous, nagios_macros contain std::string. nagios_macros now have a constructor to prevent from memset usage.